data_layout.go

```diff
@@ -0,0 +1,421 @@
+		} else if strings.HasPrefix(spec, "v") {
+			addVectorSizeAlignFromString(spec, dl)
+		} else if strings.HasPrefix(spec, "f") {
+			addFloatSizeAlignFromString(spec, dl)
+		} else if strings.HasPrefix(spec, "a") {
+			addAggAlignFromString(spec, dl)
+		} else if strings.HasPrefix(spec, "F") {
+			addFuncPtrAlignFromString(spec, dl)
+		} else if strings.HasPrefix(spec, "m") {
+			dl.mangling = ManglingStyle(strings.Split(spec, ":")[1])
+		} else if strings.HasPrefix(spec, "n") {
+			addNativeIntBitWidths(spec, dl)
+		} else if strings.HasPrefix(spec, "ni") {
+			dl.nonIntegralPointerTypes = strings.Split(strings.TrimPrefix(spec, "ni"), ":")
+		}
+	}
+	return dl
+}
+
+func addPtrSizeAlignFromString(spec string, dl *DataLayout) {
+	ptrVals := strings.Split(spec, ":")
+	ptrValsLen := len(ptrVals)
+	ptrAddSpace := uint64(0)
+	if len(ptrVals[0]) > 1 {
+		ptrAddSpace = getUInt64(strings.TrimPrefix(ptrVals[0], "p"))
+	}
+	size := getUInt64(ptrVals[1])
+	abi := getUInt64(ptrVals[2])
+	pref := abi
+	if ptrValsLen > 3 {
+		pref = getUInt64(ptrVals[3])
+	}
+	ind := size
+	if ptrValsLen == 5 {
+		ind = getUInt64(ptrVals[4])
+	}
+	dl.ptrSizeAlign[ptrAddSpace] = NewPointerSizeAlignment(ptrAddSpace, size, abi, pref, ind)
+}
+
+func addIntSizeAlignFromString(spec string, dl *DataLayout) {
+	intVals := strings.Split(spec, ":")
+	intValsLen := len(intVals)
+	size := getUInt64(strings.TrimPrefix(intVals[0], "i"))
+	abi := getUInt64(intVals[1])
+	pref := abi
+	if intValsLen == 3 {
+		pref = getUInt64(intVals[2])
+	}
+	dl.intSizeAlign[size] = NewIntegerSizeAlignment(size, abi, pref)
+}
+
+func addVectorSizeAlignFromString(spec string, dl *DataLayout) {
+	vectorVals := strings.Split(spec, ":")
+	vectorValsLen := len(vectorVals)
+	size := getUInt64(strings.TrimPrefix(vectorVals[0], "v"))
+	abi := getUInt64(vectorVals[1])
+	pref := abi
+	if vectorValsLen == 3 {
+		pref = getUInt64(vectorVals[2])
+	}
+	dl.vecSizeAlign[size] = NewVectorSizeAlignment(size, abi, pref)
+}
+
+func addFloatSizeAlignFromString(spec string, dl *DataLayout) {
+	floatVals := strings.Split(spec, ":")
+	floatValsLen := len(floatVals)
+	size := getUInt64(strings.TrimPrefix(floatVals[0], "f"))
+	abi := getUInt64(floatVals[1])
+	pref := abi
+	if floatValsLen == 3 {
+		pref = getUInt64(floatVals[2])
+	}
+	dl.floatSizeAlign[size] = NewFloatingPointSizeAlignment(size, abi, pref)
+}
+
+func addAggAlignFromString(spec string, dl *DataLayout) {
+	aggVals := strings.Split(spec, ":")
+	aggValsLen := len(aggVals)
+	abi := getUInt64(aggVals[1])
+	pref := abi
+	if aggValsLen == 3 {
+		pref = getUInt64(aggVals[2])
+	}
+	dl.aggAlign = NewAggregateAlignment(abi, pref)
+}
+
+func addFuncPtrAlignFromString(spec string, dl *DataLayout) {
+	val := strings.TrimPrefix(spec, "F")
+	var typ string
+	if strings.HasPrefix(val, "i") {
+		typ = "i"
+	} else if strings.HasPrefix(val, "n") {
+		typ = "n"
+	} else {
+		panic("Invalid Function pointer alignment type, possible options are \"i\", \"n\".")
```

Comment:
Adding the full `spec` in error string to make it easier to debug.
```suggestion
		panic(fmt.Errorf(`invalid function pointer alignment type, expected "i" or "n" prefix, got %q`, spec))
```